backend/spec/model_space_calculator_spec.rb

```diff
@@ -0,0 +1,176 @@
+require 'spec_helper'
```

Comment:
More of a general comment: this is a very small amount of tests being added compared to the size of the overall diff (~200 lines of tests against ~3000 lines of code).  Adding a controller or feature test exercising some of this functionality is probably the most efficient way to test this, given the amount of frontend view and javascript updates.